skimage/morphology/gray.py

```diff
@@ -301,30 +304,32 @@ def dilation(image, footprint=None, out=None, shift_x=False, shift_y=False):
     if out is None:
         out = np.empty_like(image)
 
+    if mode is None:
+        mode = "constant"
+        if np.issubdtype(out.dtype, bool):
+            cval = False
+        elif np.issubdtype(out.dtype, np.integer):
+            cval = np.iinfo(image.dtype).min
+        else:
+            cval = -np.inf
+
+    footprint = _shift_footprints(footprint, shift_x, shift_y)
+    footprint = pad_footprint(footprint, right=False)
+    if not mirror:
+        # Note that `ndi.grey_dilation` mirrors the footprint.
+        footprint = mirror_footprint(footprint)
```

Comment:
```suggestion
    if not mirror:
        # Note that `ndi.grey_dilation` mirrors the footprint and this
        # additional inversion should be removed in skimage2, see gh-6676.
        footprint = mirror_footprint(footprint)
```

Correct me if I'm wrong: in #6676 we settled on not silently changing the output, so that's why we keep this "bug" from earlier? Meaning this PR shouldn't close #6676; we should keep that PR open and added to the skimage2 milestone.

Reply:
See my reply to this comment in the issue report: https://github.com/scikit-image/scikit-image/issues/6676#issuecomment-1402348378